Makefile

```diff
@@ -164,13 +164,28 @@ define docker-up
 	fi
 endef
 
+docker: docker-
+
 # This implicit rule allows prefix any existing target with "docker-" to make it
 # run in docker.
 docker-%: system-checks
 	@$(call docker-up) && \
+	echo "Building docker..." && \
+	DOCKER_CONTEXT=.$(DOCKER_IMAGE_TAG)-context && \
+	mkdir -p $$DOCKER_CONTEXT && \
+	cp Dockerfile $$DOCKER_CONTEXT && \
+	echo "Calling docker build..." && \
+	docker build -t $(DOCKER_IMAGE_TAG) $$DOCKER_CONTEXT && \
+	echo docker run `echo $(DOCKER_VOLS) | xargs` -t $(DOCKER_IMAGE_TAG) /bin/bash -c 'cd /lantern && VERSION="'$$VERSION'" HEADLESS="'$$HEADLESS'" BNS_CERT_PASS="'$$BNS_CERT_PASS'" make $*' && \
+	docker run `echo $(DOCKER_VOLS) | xargs` -t $(DOCKER_IMAGE_TAG) /bin/bash -c 'cd /lantern && VERSION="'$$VERSION'" HEADLESS="'$$HEADLESS'" BNS_CERT_PASS="'$$BNS_CERT_PASS'" make $*';
+
+docker-test: system-checks
```

Comment:
@myleshorton Why does this target need to duplicate the code on lines 173 to 180?